### DISCUSSION TITLE
How do I use the EventReceiver type to get access to the underlying ResponseStream?

### DISCUSSION THREAD

**Discussion by trevorstr** [awslabs/aws-sdk-rust#1153] [General]
📦 **Crate**: `aws-sdk-bedrockagentruntime`

### Objective

I created an Amazon Bedrock Agent in the AWS Management Console, using an OpenAPI spec as input. Next, I wanted to call (invoke) the agent using the AWS SDK for Rust.

### Actual Result

The Amazon Bedrock Agent Runtime crate exposes the [InvokeAgent API](https://docs.aws.amazon.com/bedrock/latest/APIReference/API_agent-runtime_InvokeAgent.html).

The `completion` field contains the response that I need to access, however it's wrapped up in this `EventReceiver` type. How do I unwrap the `EventReceiver` to gain access to the actual `ResponseStream`? I can't find any way to accomplish this.

Here's the field I need to unwrap: https://docs.rs/aws-sdk-bedrockagentruntime/1.27.0/aws_sdk_bedrockagentruntime/operation/invoke_agent/struct.InvokeAgentOutput.html#structfield.completion

```rust
let agent_exec = agent_runtime.invoke_agent()
    .input_text("List all CloudFormation stacks, and then delete all CloudFormation stacks whose names start with a \"t\".")
    .agent_id("trevor")
    .agent_alias_id("V9QIRFYQUF")
    .session_id("session02")
    .send().await;
println!("{0:?}", agent_exec.ok().unwrap().completion); // What do I do with the EventReceiver, after referencing completion field?
```

### Expected Result

I expected there to be an easier API to retrieve results from invoking the Amazon Bedrock Agent that I created.

**Comment by ysaito1001** [comment]
Hello, to use an `EventReceiver`, you can refer to [this example](https://github.com/awslabs/aws-sdk-rust/blob/70ed857387b6b75018a126bd7435b8e323b89aae/examples/examples/transcribestreaming/src/main.rs#L92-L110). The example is for `transcribestreaming` but the same principle should apply to how to use an `EventReceiver`.
